### ISSUE TITLE
Is it stable enough?

### ISSUE THREAD

**Issue by alextes** [ultrasoundmoney/reth#1]
It would be a shame to modify Reth to support our use-cases and find out its performance is not stable enough. Let's test:
* It syncs.
* ~~We can fire a bunch of spec requests at it and it responds in timely fashion (I don't know these of the top of my head but guessing things like: what is the current head, get block 7152345 etc. would be good to craft something close to our real load).~~ It looks like we don't send any requests except for block validation requests!

Maybe Reth provides metrics we could hook into Prometheus?

**Comment by ckoopmann** [comment]
I can confirm that it does sync. Been running for a few days now in sync and everytime I checked the dashboard it looked pretty health. For the metrics we have set up prometheus and the graphana dashboard according to the reth template mentioned here:
https://paradigmxyz.github.io/reth/run/observability.html

Anything else we need to confirm its stability / observability ?

**Comment by alextes** [comment]
Not really. I used to think so but there's actually no endpoints we use except the block validation one which is why I scratched out the line above.

Something we could do as a minimum test still. Ask our node for the head block hash a couple times a second, ask a trusted provider like Alchemy the same, analyze if there are any unusual discrepancies. Wouldn't be weird if there's some, it is decentralized tech, but there shouldn't be many.

Another test which matters for us: for a shorter period of time we could also test head update latency, asking for the head maybe every 50ms or so, to check that the node updates its idea of the head in timely fashion compared to service like Alchemy.

But that's plenty as to stability. Also seems possible to do in a couple hours of scripting.

As a final idea, perhaps others have already done research like this? No need to test these things if the Reth team or others have already confirmed the node doesn't just sync but stays in sync well and responds to RPC calls without delay.

**Comment by gakonst** [comment]
The node syncs well atm and has been stable for most people (absent of some infrequent hardware-specific issues). We haven't found any consensus deviations so far. Stability for us critically means we don't break the DB format or 3rd party itnegrations at the code level, on top of having excellent SLA and no dirty states on SIGINT and so on. Happy to dive in more if you have further Qs/areas for digging.

> Another test which matters for us: for a shorter period of time we could also test head update latency, asking for the head maybe every 50ms or so, to check that the node updates its idea of the head in timely fashion compared to service like Alchemy.

Happy to help with this, block notification performance is something we really care about and would like to measure more. I think that we have comparable times with Geth
